tools/linter/adapters/import_linter.py

```diff
@@ -0,0 +1,391 @@
+            "socketserver",
+            "spwd",
+            "sqlite3",
+            "sre_compile",
+            "sre_constants",
+            "sre_parse",
+            "ssl",
+            "stat",
+            "statistics",
+            "string",
+            "stringprep",
+            "struct",
+            "subprocess",
+            "sunau",
+            "symbol",
+            "symtable",
+            "sys",
+            "sysconfig",
+            "syslog",
+            "tabnanny",
+            "tarfile",
+            "telnetlib",
+            "tempfile",
+            "termios",
+            "test",
+            "textwrap",
+            "this",
+            "threading",
+            "time",
+            "timeit",
+            "tkinter",
+            "token",
+            "tokenize",
+            "trace",
+            "traceback",
+            "tracemalloc",
+            "tty",
+            "turtle",
+            "turtledemo",
+            "types",
+            "typing",
+            "unicodedata",
+            "unittest",
+            "urllib",
+            "uu",
+            "uuid",
+            "venv",
+            "warnings",
+            "wave",
+            "weakref",
+            "webbrowser",
+            "winreg",
+            "winsound",
+            "wsgiref",
+            "xdrlib",
+            "xml",
+            "xmlrpc",
+            "xxsubtype",
+            "zipapp",
+            "zipfile",
+            "zipimport",
+            "zlib",
+            "zoneinfo",
+        ]
+    )
+
+# Add the allowed third party libraries.
+_MODULE_NAME_ALLOW_LIST.update(
+    [
+        "torch",
+        "sympy",
+        "torch_xla",
+        "_pytest",
+        "functorch",
+        "einops",
+        "the",
+        "libfb",
+        "typing_extensions",
+        "triton",
+        "numpy",
+        "torchrec",
+        "tabulate",
+        "optree",
+        "tvm",
+    ]
+)
+
+
+def check_file(filename: str) -> List[LintMessage]:
+    with open(filename) as f:
+        lines = f.readlines()
+
+    # The pattern: from/import word_that_doesn't_start_with_dot
+    pattern = re.compile(r"^(?:import|from)\s+([a-zA-Z_][\w]*)")
+
+    lint_messages = []
+    for line_number, line in enumerate(lines):
+        line_number += 1
+        line = line.lstrip()
+        match = pattern.search(line)
+        if match:
+            module_name = match.group(1)
+            if module_name not in _MODULE_NAME_ALLOW_LIST:
+                msg = LintMessage(
+                    path=filename,
+                    line=line_number,
+                    char=None,
+                    code="IMPORT",
+                    severity=LintSeverity.ERROR,
+                    name="Disallowed import",
+                    original=None,
+                    replacement=None,
+                    description=f"""
+importing from {module_name} is not allowed, if you believe there's a valid
+reason, please add it to _MODULE_NAME_ALLOW_LIST in {CURRENT_FILE_NAME}
```

Comment:
Nit: let's state the perils of importing a third-party module. "Please do not import third-party modules in PyTorch unless they're explicitly requirements of PyTorch. Imports of a third-party library may have side effects and other unintentional behavior".

Reply:
We can also state the remedy. "If you're just checking if a module exists, use sys.modules.get("torchrec")"

Reply:
Updated.